src/SetManyCapableTrait.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace Dhii\Data\Object;
+
+use Dhii\Util\String\StringableInterface as Stringable;
+use OutOfRangeException;
+use stdClass;
+use Traversable;
+use Exception as RootException;
+use InvalidArgumentException;
+
+/**
+ * Functionality for data assignment.
+ *
+ * @since [*next-version*]
+ */
+trait SetManyCapableTrait
+{
+    /**
+     * Assign data.
+     *
+     * @since [*next-version*]
+     *
+     * @param array|stdClass|Traversable $data The data to set. Existing keys will be overwritten.
+     *                                         The rest of the data remains unaltered.
+     *
+     * @throws InvalidArgumentException If the list of keys is invalid.
+     * @throws OutOfRangeException      If trying to set data for an invalid key.
+     * @throws RootException            If a problem occurs while writing data.
+     */
+    protected function _setMany($data)
+    {
+        $data = $this->_normalizeIterable($data);
+
+        foreach ($data as $_key => $_value) {
+            try {
+                $this->_setData($_key, $_value);
+            } catch (InvalidArgumentException $e) {
+                throw $this->_createOutOfRangeException(
```

Comment:
Surely this must fit in 1 line. Can't it? 🤣

Reply:
Prolly not

Reply:
I actually checked before writing this comment. It does 😛

Reply:
Anything can fit in one line. But should it?

Reply:
Oh crap. You're right. Should it? Ok, you have my go ahead. Fit everything into 1 line.

Reply:
I should probably follow up on that. That was a joke. Please don't obfuscate ✌️ 

I only brought this up because I like clean code. It doesn't bother me _that_ much.

Reply:
It was you who started writing these things in the multi-line fashion. Since I'm all about standards, I decided that all exception factory methods that specify the 4th parameter should be written this way, regardless of length.